### PR TITLE
CORTX-32078 - Updated ceph execution script

### PR DESCRIPTION
**Pull request by ketanarlulkar**
# Problem Statement
- Adding support for ceph test execution from Jenkins regression job

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [x] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by ketanarlulkar** [comment]
Execution Logs: [QA Jobs » QA_R2_Single_Runner-debug #27 Console [Jenkins] (seagate.com)](https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner-debug/27/console)

**Review by ravindrachoudhari — APPROVED** [review]

File: ceph_s3tests_runner.py [hunk 1]
```diff
@@ -60,19 +76,158 @@ def get_tests_from_te(jira_obj, args, test_type=None):
...
+        config.write(configfile)
+
+
+def check_or_create_accounts():
+    """Create IAM accounts required for ceph test execution if accounts does not exit already"""
+    iam_users_sections = ["s3 main", "s3 alt", "s3 tenant", "iam"]
+    csm_obj = csm_api_factory("rest")
+
+    resp = csm_obj.list_iam_users_rgw()
+    if resp.status_code != HTTPStatus.OK:
```
Comment (sonalk0209) [inline ceph_s3tests_runner.py:115]: will this give u error even if no IAM users are created?

File: scripts/ceph_s3_tests/cortx_rgw_template.conf [hunk 2]
```diff
@@ -10,75 +10,81 @@ port = 80
...
-display_name = <S3_MAIN_USER_NAME>
+display_name = S3_MAIN_USER
 
 # main user_idname set in vstart.sh
-user_id = <S3_MAIN_USER_ID>
+user_id = S3_MAIN_USER
 
 # main email set in vstart.sh
-email = s3test@seagate.com
+email = s3_main_user@seagate.com
```
Comment (sonalk0209) [inline scripts/ceph_s3_tests/cortx_rgw_template.conf:30]: any security risk here giving mail id directly? @ravindrachoudhari

File: ceph_s3tests_runner.py (hunk 1 above)
Comment (ketanarlulkar) [reply ceph_s3tests_runner.py:115]: Even if no users, response status code should be 200 (OK)

File: ceph_s3tests_runner.py [hunk 3]
```diff
@@ -60,19 +76,158 @@ def get_tests_from_te(jira_obj, args, test_type=None):
...
+def get_test_data(jira_obj, test_ticket):
+    """Get TEST data from TEST JIRA ticket"""
+    test = jira_obj.get_issue_details(test_ticket)
+    test_definition = test.fields.customfield_20984
+    summary = test.fields.summary
+    components = test.fields.components
+    test_team = components[0].name if components and isinstance(components, list) else "CortxQA"
+    feature = test.fields.customfield_21087.value if test.fields.customfield_21087 else "None"
+    dr_id = test.fields.customfield_22882 if test.fields.customfield_22882 else ['DR-0']
+    feature_id = test.fields.customfield_22881 if test.fields.customfield_22881 else ['F-0']
```
[the remark this answers is not in the document]
Comment (ketanarlulkar) [reply ceph_s3tests_runner.py:149]: These are the same default values which are used in testrunner as well. So, using those for now. This field is compulsory to create entry in DB.

File: ceph_s3tests_runner.py [hunk 4]
```diff
@@ -60,19 +76,158 @@ def get_tests_from_te(jira_obj, args, test_type=None):
...
+        raise EnvironmentError(f"Definition field for {test} JIRA is empty.")
+    return test_to_run
+
+
+def setup_configurations(args):
+    """Setup host:port and security protocol"""
+    setup_details = ClientConfig(get_db_credential()).get_setup_details(args.target)
+    if not os.path.exists(CONFIG_FILE):
+        LOGGER.info("Coping default config file")
+        shutil.copyfile("scripts/ceph_s3_tests/cortx_rgw_template.conf", CONFIG_FILE)
```
[the remark this answers is not in the document]
Comment (ketanarlulkar) [reply ceph_s3tests_runner.py]: Removed if replacing file directly.

File: scripts/ceph_s3_tests/cortx_rgw_template.conf (hunk 2 above)
Comment (ravindrachoudhari) [reply scripts/ceph_s3_tests/cortx_rgw_template.conf:30]: I don't think so.

**Review by apoorva-rao — APPROVED** [review]
LGTM

**Review by sonalk0209 — APPROVED** [review]
Pls add collections logs

File: scripts/ceph_s3_tests/cortx_rgw_template.conf [hunk 5]
```diff
@@ -10,75 +10,81 @@ port = 80
...
-secret_key = <S3_TNT_SECRET_KEY>
+secret_key = S3_IAM_SECRET_KEY
 
 #display_name from vstart.sh
-display_name = s3test3
\ No newline at end of file
+display_name = S3_IAM_USER
+
+#iam email set in vstart.sh
+email = s3_iam_user@example.com
```
Comment (apurwamohite) [inline scripts/ceph_s3_tests/cortx_rgw_template.conf:90]: shouldn't this be a string?

**Review by apurwamohite — APPROVED** [review]

File: scripts/ceph_s3_tests/cortx_rgw_template.conf (hunk 5 above)
Comment (ketanarlulkar) [reply scripts/ceph_s3_tests/cortx_rgw_template.conf:90]: It is string, quotes are not mandatory

**Comment by ketanarlulkar** [comment]
Collection Logs: Attached to JIRA [here](https://jts.seagate.com/browse/CORTX-33485?focusedCommentId=2421358&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2421358)

**Pull request merged**